cmd/shared/README.md

```diff
@@ -0,0 +1,54 @@
+# C shared library
+
+This is an experiment in exposing the internals of Secretless as a C shared library.
+Static would be best but for now this needs dynamic linking since some Secretless components use CGO.
+
```

Comment:
It should include something about its state here - I'd call it `pre-alpha` bc it has no tests, wdyt?

Reply:
Sounds good